networkx/algorithms/community/label_propagation.py

```diff
@@ -0,0 +1,120 @@
+#-*- coding: utf-8 -*-
+#    Copyright (C) 2015 
+#    All rights reserved.
+#    BSD license.
+
+
+"""
+Semi-synchronous label propagation community detection algorithm.
+"""
+
+from collections import Counter
+from networkx.utils.decorators import not_implemented_for
+
+__all__ = ["label_propagation_communities"]
+
+@not_implemented_for('directed')
+def label_propagation_communities(G):
+    """Finds communities in graph using the label propagation method[1]_. This
+    method uses the diffusion of information in the network to identify
+    communities. Not implemented for directed graphs.
+    
+    Parameters
+    ----------
+    G : graph
+    An undirected NetworkX graph.      
+    
+    Returns
+    -------
+    communities : list
+    List of the communities (sets)
+    
+    Raises
+    ------
+    NetworkXNotImplemented
+       If the graph is directed
+  
+    References
+    ----------
+    .. [1] Cordasco, G., & Gargano, L. (2010, December). Community detection 
+       via semi-synchronous label propagation algorithms. In Business 
+       Applications of Social Network Analysis (BASNA), 2010 IEEE International 
+       Workshop on (pp. 1-8). IEEE.
+    """
+    coloring = _color_network(G)
+    # Create a unique label for each node in the graph
+    labeling = {v: k for k, v in enumerate(G)}
+    while not _labeling_complete(labeling, G):
+        # Update the labels of every node.
+        for color, nodes in coloring.items():
+            for n in nodes:
+                _update_label(n , labeling, G)
+        
+    for label in set(labeling.values()):
+        yield set((x for x in labeling if labeling[x] == label))
+        
+def _color_network(G):
```

Comment:
Instead of writing your own function here, can you use `networkx.coloring.greedy_coloring` instead?